Sources/Auth/Core/JWTDecoder.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  JWTDecoder.swift
+//  PovioKit
+//
+//  Created by Borut Tomazin on 10/1/2023.
+//  Copyright © 2023 Povio Inc. All rights reserved.
+//
+
+import Foundation
+
+/// JWTDecoder for decoding JSON Web Tokens (JWT) tokens
+/// Inspired by https://github.com/auth0/jwt-decode
+public struct JWTDecoder {
```

Comment:
We could write some tests for this implementation.

Reply:
Didn't get there yet. A separate PR is incoming.